build: Remove obsolete autoconf macros

jelly opened the pull request:
Since autoconf 2.70 all obsolete macro warnings are turned on by default and prints the following message when running `./autogen.sh`.

```
configure.ac:45: warning: The macro `AC_PROG_CC_C99' is obsolete.
configure.ac:45: You should run autoupdate.
./lib/autoconf/c.m4:1659: AC_PROG_CC_C99 is expanded from...
configure.ac:45: the top level
configure.ac:47: warning: The macro `AC_ISC_POSIX' is obsolete.
configure.ac:47: You should run autoupdate.
./lib/autoconf/specific.m4:550: AC_ISC_POSIX is expanded from...
configure.ac:47: the top level
configure.ac:48: warning: The macro `AC_HEADER_STDC' is obsolete.
configure.ac:48: You should run autoupdate.
./lib/autoconf/headers.m4:704: AC_HEADER_STDC is expanded from...
configure.ac:48: the top level
configure.ac:95: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:95: You should run autoupdate.
./lib/autoconf/general.m4:204: AC_HELP_STRING is expanded from...
configure.ac:95: the top level
configure.ac:136: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:136: You should run autoupdate.
./lib/autoconf/general.m4:204: AC_HELP_STRING is expanded from...
configure.ac:136: the top level
configure.ac:163: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:163: You should run autoupdate.
./lib/autoconf/general.m4:204: AC_HELP_STRING is expanded from...
./lib/autoconf/general.m4:1553: AC_ARG_WITH is expanded from...
configure.ac:163: the top level
configure.ac:181: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:181: You should run autoupdate.
./lib/autoconf/general.m4:204: AC_HELP_STRING is expanded from...
configure.ac:181: the top level
configure.ac:216: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:216: You should run autoupdate.
./lib/autoconf/general.m4:204: AC_HELP_STRING is expanded from...
./lib/autoconf/general.m4:1553: AC_ARG_WITH is expanded from...
configure.ac:216: the top level
configure.ac:298: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:298: You should run autoupdate.
./lib/autoconf/general.m4:204: AC_HELP_STRING is expanded from...
configure.ac:298: the top level
configure.ac:368: warning: The macro `AC_HELP_STRING' is obsolete.

```

martinpitt commented on configure.ac:
```diff
@@ -92,7 +92,7 @@ AC_SUBST(COCKPIT_CFLAGS)
 
 # bridge with optional polkit
-AC_ARG_ENABLE(polkit, AC_HELP_STRING([--disable-polkit], [Disable usage of polkit]),,enable_polkit=yes)
+AC_ARG_ENABLE(polkit, AS_HELP_STRING([--disable-polkit], [Disable usage of polkit]),,enable_polkit=yes)
```
Ack, http://www.gnu.org/software/autoconf/manual/autoconf-2.67/html_node/Pretty-Help-Strings.html already documents the `AS_*` variant. Do you mind adding a doc URL to the commit message for this?

martinpitt commented on configure.ac:
```diff
@@ -44,8 +44,6 @@ AM_PROG_AR
 AC_PROG_CC_C99
 AC_PROG_RANLIB
-AC_ISC_POSIX
-AC_HEADER_STDC
```
Commit message has "Interacitve" typo.

jelly replied:
Fixed

martinpitt reviewed (changes requested):
Nice, thanks! Two things:

 - As we will build with 2.69 for at least the near future, is it possible to enable warnings with 2.69, so that we can see them in our build?
 - Please prefix your commits with "build:"; we usually have a page/topic prefix such as "networking", "build", or "test" which makes them easier to scan when e.g. looking at `git shortlog` for drafting release summaries, or filtering commits.

jelly commented:
> Nice, thanks! Two things:
> 
> * As we will build with 2.69 for at least the near future, is it possible to enable warnings with 2.69, so that we can see them in our build?

Added `--warnings obsolete` to autoreconf which should warn with autoconf < 2.70

> * Please prefix your commits with "build:"; we usually have a page/topic prefix such as "networking", "build", or "test" which makes them easier to scan when e.g. looking at `git shortlog` for drafting release summaries, or filtering commits.

Done and squashed all commits.

martinpitt reviewed (approved):
Very cool, thanks for the cleanup! Triggered tests.

martinpitt commented:
FTR: We [now see two warnings](https://github.com/cockpit-project/cockpit/pull/16191/checks?check_run_id=3239764792#step:4:52):
```
configure.ac:502: warning: AC_OUTPUT should be used without arguments.
configure.ac:502: You should run autoupdate.
configure.ac:502: warning: AC_OUTPUT should be used without arguments.
configure.ac:502: You should run autoupdate.
```
Previously it was clean. But this is a good reminder to fix that last bit as well. Maybe you want to look at this? (Not urgent/important)

jelly commented:
> FTR: We [now see two warnings](https://github.com/cockpit-project/cockpit/pull/16191/checks?check_run_id=3239764792#step:4:52):
> 
> ```
> configure.ac:502: warning: AC_OUTPUT should be used without arguments.
> configure.ac:502: You should run autoupdate.
> configure.ac:502: warning: AC_OUTPUT should be used without arguments.
> configure.ac:502: You should run autoupdate.
> ```
> 
> Previously it was clean. But this is a good reminder to fix that last bit as well. Maybe you want to look at this? (Not urgent/important)

Noticed those warnings as well, I can take a look later this week/next week
